File: Backend/SOP/SOP/Entities/ComputerPart.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SOP.Entities
{
    public class ComputerPart
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("PartGroup.Id")]
        public int PartGroupId { get; set; }

        [Column(TypeName = "nvarchar(50)")]
        public string SerialNumber { get; set; }

        [Column(TypeName = "nvarchar(50)")]
        public string ModelNumber { get; set; }

        public PartGroup PartGroup { get; set; }
        public Computer_ComputerPart Computer_ComputerPart { get; set; }

    }
}

File: Frontend/src/app/models/computerPart.ts
import { Computer_ComputerPart } from "./computer_ComputerPart";
import { PartGroup } from "./partGroup";

/*
*This is the object structure of the computerPart object.
*It is used to store it in the database from frontend to backend. 
*/

export interface ComputerPart {
    id: number,
    partGroupId: number,
    serialNumber: string,
    modelNumber: string,
    // *get the partGroup object from the partGroup.ts file
    group?: PartGroup,
    // *get the computer_ComputerPart object from the computer_ComputerPart.ts file
    computer_ComputerPart?: Computer_ComputerPart[]
}
